### ISSUE TITLE
ConvertTo-DSCObject doesn't deliver proper object when Import-DscResource module doesn't exists

### ISSUE THREAD

**Issue by PrzemyslawKlys** [microsoft/DSCParser#18]
If you try to process DSC file while the Import-DSCResource references module that doesn't exists (in my case I had newer version) it would not deliver proper object

![image](https://user-images.githubusercontent.com/15063294/154354643-4f405806-6785-41f9-842b-0ab629003ae9.png)

Object returned:

![image](https://user-images.githubusercontent.com/15063294/154354824-a6465607-475b-4b22-9609-ab78f043ecc7.png)

```powershell

$Test = ConvertTo-DSCObject -Path C:\Support\GitHub\Testimo\Ignore\m365\M365TenantConfig.ps1 -IncludeComments $true
$Test | Format-Table
```

**Comment by PrzemyslawKlys** [comment]
The reason for this is the AST is different. Would require updating logic

![image](https://user-images.githubusercontent.com/15063294/154755773-34e1faab-e142-407a-8b42-ef3d25f224b5.png)

**Comment by andikrueger** [comment]
Within Microsoft365DSC there is logic to remove the required version entirely. This impacts the process if there are several versions of a module present. https://github.com/microsoft/Microsoft365DSC/blob/Dev/Modules/Microsoft365DSC/Modules/M365DSCReport.psm1#L33-L43

I really like the idea to show an information/warning about a miss match of versions.

**Comment by PrzemyslawKlys** [comment]
I think removing is one thing, but there should be logic to maybe not setting it up first.

**Comment by andikrueger** [comment]
I totally agree with you. Removing this part is just going around the problem - and this also cause some additional issues: see https://github.com/microsoft/Microsoft365DSC/issues/1725#issuecomment-1047227465
